vulcanai/utils.py

```diff
@@ -419,3 +430,163 @@ def stitch_datasets(df_list, on, index_list=None):
+                                          convert_to_class=False)
+    class_prediction = get_class(raw_prediction)
+
+    confusion_matrix = get_confusion_matrix(
+        prediction=class_prediction,
+        truth=test_y
+    )
+
+    tp = np.diagonal(confusion_matrix).astype('float32')
+    tn = (np.array([np.sum(confusion_matrix)] *
+                   confusion_matrix.shape[0]) -
+          confusion_matrix.sum(axis=0) -
+          confusion_matrix.sum(axis=1) + tp).astype('float32')
+    # sum each column and remove diagonal
+    fp = (confusion_matrix.sum(axis=0) - tp).astype('float32')
+    # sum each row and remove diagonal
+    fn = (confusion_matrix.sum(axis=1) - tp).astype('float32')
+
+    sens = np.nan_to_num(tp / (tp + fn))  # recall
+    spec = np.nan_to_num(tn / (tn + fp))
+    dice = 2 * tp / (2 * tp + fp + fn)
+    ppv = np.nan_to_num(tp / (tp + fp))  # precision
+    npv = np.nan_to_num(tn / (tn + fn))
+    accuracy = np.sum(tp) / np.sum(confusion_matrix)
+    f1 = np.nan_to_num(2 * (ppv * sens) / (ppv + sens))
+
+    print ('{} test\'s results'.format(network.name))
+
+    print ('TP:'),
+    print (tp)
+
+
+    print ('FP:'),
+    print (fp)
+
+
+    print ('TN:'),
+    print (tn)
+
+
+    print ('FN:'),
+    print (fn)
+
+
+    print ('\nAccuracy: {}'.format(accuracy))
+    temp_df["Accuracy"] = [accuracy]
+
+    print ('Sensitivity:'),
+    print(round_list(sens, decimals=3))
+
+
+    print ('Specificity:'),
+    print(round_list(spec, decimals=3))
+
+
+    print ('DICE:'),
+    print(round_list(dice, decimals=3))
+    print ('\tAvg. DICE: {:.4f}'.format(np.average(dice)))
+
+    print ('Positive Predictive Value:'),
+    print(round_list(ppv, decimals=3))
+
+
+    print ('Negative Predictive Value:'),
+    print(round_list(npv, decimals=3))
+
+
+    print ('f1-score:'),
+    print(round_list(f1, decimals=3))
+
+    all_class_auc = []
+    for i in range(network.num_classes):
+        if network.num_classes == 1:
+            fpr, tpr, thresholds = metrics.roc_curve(test_y,
+                                                     raw_prediction,
+                                                     pos_label=1)
+        else:
+            fpr, tpr, thresholds = metrics.roc_curve(test_y,
+                                                     raw_prediction[:, i],
+                                                     pos_label=i)
+
+        auc = metrics.auc(fpr, tpr)
+        all_class_auc += [auc]
+
+    print ('Average AUC: : {:.4f}'.format(np.average(all_class_auc)))
+
+
+    if test_df is None:
+        test_df = pd.DataFrame(columns=["Feature", "Value", "TP", "FP", "TN", "FN",
+                                        "Accuracy", "Sensitivity", "Specificity", "PPV",
+                                        "NPV", "AUC"])
+
+    temp_df["Feature"] = [feature]
+    temp_df["Value"] = [value]
+    temp_df["AUC"] = [np.average(all_class_auc)]
+    temp_df["NPV"] = [npv[0]]
```

Comment:
I would like to generalize these metrics to their entire contents so we collect info from all classes at once if possible.